packages/xchain-ethereum/src/client.ts

```diff
@@ -119,67 +105,71 @@ export default class Client implements XChainClient, EthereumClient {
   /**
    * Getters
    */
+
+  getNetwork(): XChainNetwork {
+    return ethNetworkToXchains(this.network)
+  }
+
   getAddress(): Address {
-    return this._address
+    if (!this.address) {
+      throw new Error('Phrase must be provided')
```

Comment:
Error message needs to be updated to tell an user, why a phrase needs to be set to have an address.

Also we should add a comment to each function, which might throw an error. Without this information, nobody knows about it and can't handle a call of such function in a right way (e.g. by using `try / catch`). Also our generated documentation will profit from it.

We don't have such comments in other clients yet, but let's start doing this in `xchain-eth`. Here is an example for `getAddress`

```
 * @throws Error
 * Thrown if phrase has not been set before. A phrase is needed to create a wallet and to derive an address from it.
```

See[ @throws at tsdoc.org](https://tsdoc.org/pages/tags/throws/)